### PR TITLE
Add more time syscalls and fix `touch` command in busybox

### PR DESCRIPTION
**Pull request by yunwei37**
## Add more time syscalls

- [x] Add more time related syscalls:
  - `time(  )`
  - `gettimeofday(  )`
  - `gettusage( )`
  - `times( )`
- [x] update document for these syscalls
- [x] unit test

note: these syscalls is not very complete, but support basic function of getting time ( the same in rCore ).

## related: fix `touch` command

- [x] use `touch` command in busybox, need to fix `utimensat` syscall
  - touch can create file successfully!
- [x] update document
- [x] test with busybox

## sides

- add fcntl consts in `linux-object`
- [x] add more test with current commands in busybox;

### CODE REVIEW

File: linux-object/src/fs/fcntl.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//! consts for fctnl
```
Comment (wangrunji0408) [inline linux-object/src/fs/fcntl.rs:1]: note: some of the consts in this file should be converted into `enum` or `bitflags` in the future.

File: linux-syscall/src/time.rs [hunk 2]
```diff
@@ -1,32 +1,199 @@
...
+impl TimeVal {
+    /// create TimeVal
+    pub fn new() -> TimeVal {
+        TimeSpec::new().into()
+    }
+}
+
+impl TimeSpec {
+    /// create TimeSpec
+    pub fn new() -> TimeSpec {
```
Comment (wangrunji0408) [inline linux-syscall/src/time.rs:123]: Maybe a better name: `now`.
Or `impl From<Duration> for TimeSpec`, then `TimeSpec::from(timer_now())`.
Which one is better?

File: linux-syscall/src/time.rs [hunk 3]
```diff
@@ -1,32 +1,199 @@
...
+        buf.write(new_buf)?;
 
+        info!("tick: {:?}", tick);
+        Ok(tick as usize)
+    }
+}
+
+impl TimeVal {
+    /// create TimeVal
+    pub fn new() -> TimeVal {
```
Comment (wangrunji0408) [inline linux-syscall/src/time.rs:116]: Maybe a better name: `now`

File: linux-syscall/src/time.rs [hunk 4]
```diff
@@ -1,32 +1,199 @@
...
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Default for TimeSpec {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```
Comment (wangrunji0408) [inline linux-syscall/src/time.rs:179]: Intuitively the default value of `TimeVal` or `TimeSpec` should be 0? Just like `Duration` does.

File: linux-syscall/src/time.rs [hunk 5]
```diff
@@ -1,32 +1,199 @@
...
+
+/// Tms for times()
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct Tms {
+    tms_utime: u64,  /* user time */
+    tms_stime: u64,  /* system time */
+    tms_cutime: u64, /* user time of children */
+    tms_cstime: u64, /* system time of children */
+}
```
Comment (wangrunji0408) [inline linux-syscall/src/time.rs:199]:
```suggestion
pub struct Tms {
    /// user time
    utime: u64,
    /// system time
    stime: u64,
    /// user time of children
    cutime: u64,
    /// system time of children
    cstime: u64,
}
```

File: linux-loader/src/main.rs [hunk 6]
```diff
@@ -122,5 +148,13 @@ mod tests {
     #[async_std::test]
     async fn test_pipe() {
         test("/bin/testpipe1").await;
+        let string = fs::read_to_string("../rootfs/testpipe.txt").unwrap();
+        assert_eq!(string, String::from("hello pipe\n"));
+        test("/bin/busybox rm testpipe.txt").await;
+    }
```
Comment (wangrunji0408) [inline linux-loader/src/main.rs:154]: I think the validation procedure should be moved into the test program, then it indicates test result (pass or failed) through the return value (0 or not).

**Review by wangrunji0408 — CHANGES_REQUESTED** [review]

File: linux-loader/src/main.rs (hunk 6 above)
Comment (yunwei37) [reply linux-loader/src/main.rs:154]: Thank you， I will fix these. ( It should be like this...

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**